src/autoscaler/metricscollector/cmd/metricscollector/main.go

```diff
@@ -113,27 +113,28 @@ func main() {
 		os.Exit(1)
 	}
 
-	consulClient, err := consuladapter.NewClientFromUrl(conf.Lock.ConsulClusterConfig)
-	if err != nil {
-		logger.Fatal("new consul client failed", err)
+	members := grouper.Members{
+		{"collector", collectServer},
+		{"http_server", httpServer},
 	}
 
-	serviceClient := metricscollector.NewServiceClient(consulClient, mcClock)
+	if conf.Lock.ConsulClusterConfig != "" {
+		consulClient, err := consuladapter.NewClientFromUrl(conf.Lock.ConsulClusterConfig)
+		if err != nil {
+			logger.Fatal("new consul client failed", err)
+		}
 
-	lockMaintainer := serviceClient.NewMetricsCollectorLockRunner(
-		logger,
-		generateGUID(logger),
-		conf.Lock.LockRetryInterval,
-		conf.Lock.LockTTL,
-	)
+		serviceClient := metricscollector.NewServiceClient(consulClient, mcClock)
+		lockMaintainer := serviceClient.NewMetricsCollectorLockRunner(
+			logger,
+			generateGUID(logger),
+			conf.Lock.LockRetryInterval,
+			conf.Lock.LockTTL,
+		)
 
-	registrationRunner := initializeRegistrationRunner(logger, consulClient, conf.Server.Port, mcClock)
-
-	members := grouper.Members{
-		{"lock-maintainer", lockMaintainer},
-		{"collector", collectServer},
-		{"http_server", httpServer},
-		{"registration", registrationRunner},
+		registrationRunner := initializeRegistrationRunner(logger, consulClient, conf.Server.Port, mcClock)
+		members = append(members, grouper.Member{"lock-maintainer", lockMaintainer})
```

Comment:
Lock maintainer should be the first one in the members otherwise collector and http_server would start even when the lock is not achieved.